pkg/types/types.go

```diff
@@ -72,6 +73,14 @@ type Address []byte
 // upkeep at a block number
 type UpkeepKey []byte
```

Comment:
If we move the implementation to the chainlink repo, how should we deal with instantiating an UpkeepKey implementation within the this repo, e.g.: https://github.com/smartcontractkit/ocr2keepers/blob/main/pkg/chain/evmreports.go#L186

Reply:
Does it maybe make sense to have the UpkeepKey interface and implementations live in libocr? 🤔

Reply:
To allow the simulators to continue functioning, it might make sense to put the implementation in `pkg/chain` for now. This way the `chainlink` repo can access it, implementations can be chain specific, and the simulators can use the implementations. 

`libocr` should only be general OCR stuff. The UpkeepKey is specific to the plugin.